src/main/java/sirius/biz/jobs/batch/file/FileImportJob.java

```diff
@@ -196,23 +203,81 @@ protected void executeForArchive(String filename, FileHandle fileHandle) throws
         process.log(ProcessLog.info().withNLSKey("FileImportJob.importingZipFile"));
 
         AtomicInteger filesImported = new AtomicInteger();
+        if (entriesToExtract.isEmpty()) {
+            extractAllEntries(filename, fileHandle, filesImported::incrementAndGet);
+        } else {
+            extractEntriesFromList(filename, fileHandle, filesImported::incrementAndGet);
+        }
+
+        if (filesImported.get() == 0) {
+            throw Exceptions.createHandled().withNLSKey("FileImportJob.noZippedFileFound").handle();
+        }
+    }
+
+    private void extractAllEntries(String filename, FileHandle fileHandle, Runnable counter) {
         extractor.extractAll(filename, fileHandle.getFile(), null, file -> {
             if (executeForEntry(file)) {
-                filesImported.incrementAndGet();
+                counter.run();
             }
         });
+    }
 
-        if (filesImported.get() == 0) {
-            throw Exceptions.createHandled().withNLSKey("FileImportJob.noZippedFileFound").handle();
-        }
+    private void extractEntriesFromList(String filename, FileHandle fileHandle, Runnable counter) {
+        entriesToExtract.forEach((fileName, fileRequired) -> {
+            Monoflop entryFound = Monoflop.create();
+            extractor.extractAll(filename, fileHandle.getFile(), entryName -> {
+                return entryName.equals(fileName);
+            }, file -> {
+                if (executeForEntry(file)) {
+                    counter.run();
+                }
+                entryFound.toggle();
+            });
+            if (entryFound.isToggled()) {
+                return;
+            }
+
+            if (Boolean.TRUE.equals(fileRequired)) {
+                throw Exceptions.createHandled()
+                                .withNLSKey("FileImportJob.requiredFileNotFound")
+                                .set("filename", fileName)
+                                .handle();
+            } else {
+                process.log(ProcessLog.info()
+                                      .withNLSKey("FileImportJob.requiredFileNotFound")
+                                      .withContext("filename", fileName));
+            }
+        });
+    }
+
+    /**
+     * Specifies specific entries to extract from an archive.
+     * <p>
+     * Override this method in order to filter specific entries.
+     *
+     * @see #addEntryFilter(String, boolean)
+     */
+    protected void defineEntriesToExtract() {
```

Comment:
```suggestion
    protected void defineEntriesToExtract(BiConsumer<String,Boolean> entryConsumer) {
```

Maybe directly provide the consumer here instead via addEntryFilter - to avoid that addEntryFilter is called elsehwere...